addons/test_mail/static/tests/attachment_view.test.js

```diff
@@ -0,0 +1,137 @@
+            const doc = popoutIframe.contentDocument;
+            const originalWrite = doc.write;
+            doc.write = (content) => {
+                // This avoids duplicating the test script in the popoutWindow
+                const sanitizedContent = content.replace(
+                    /<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi,
+                    ""
+                );
+                originalWrite.call(doc, sanitizedContent);
+            };
+            return doc;
+        },
+        get body() {
+            return popoutWindow.document.querySelector(".o_popout_wrapper");
+        },
+        close: () => {
+            popoutWindow.closed = true;
+            popoutIframe.remove(popoutWindow.body);
+        },
+        contains: async (selector) => {
+            await animationFrame();
+            await waitUntil(() => popoutWindow.body);
+            const target = popoutWindow.body.querySelector(selector);
+            expect(target).toBeDisplayed();
+            return target;
+        },
+        click: async (selector) => {
+            const target = await popoutWindow.contains(selector);
+            hootClick(target);
+        },
+    };
+    patchWithCleanup(browser, {
+        open: () => {
+            queryOne(".o_popout_holder").append(popoutIframe);
+            return popoutWindow;
+        },
+    });
+
+    const pyEnv = await startServer();
+    const recordId = pyEnv["mail.test.simple.main.attachment"].create({
+        display_name: "first partner",
+        message_attachment_count: 2,
+    });
+    const attachmentIds = pyEnv["ir.attachment"].create([
+        {
+            mimetype: "image/jpeg",
+            res_id: recordId,
+            res_model: "mail.test.simple.main.attachment",
+        },
+        {
+            mimetype: "application/pdf",
+            res_id: recordId,
+            res_model: "mail.test.simple.main.attachment",
+        },
+    ]);
+    pyEnv["mail.message"].create({
+        attachment_ids: attachmentIds,
+        model: "mail.test.simple.main.attachment",
+        res_id: recordId,
+    });
+    registerArchs({
+        "mail.test.simple.main.attachment,false,form": `
+                <form string="Test document">
+                    <div class="o_popout_holder"/>
+                    <sheet>
+                        <field name="name"/>
+                    </sheet>
+                    <div class="o_attachment_preview"/>
+                    <chatter/>
+                </form>`,
+    });
+
+    patchUiSize({ size: SIZES.XXL });
+    await start();
+    await openFormView("mail.test.simple.main.attachment", recordId);
+    await click(".o_attachment_preview .o_move_next");
+    await contains(".o_attachment_preview img");
+    await click(".o_attachment_preview .o_move_previous");
+    await contains(".o_attachment_preview iframe");
+
+    await click(".o_attachment_preview .o_attachment_control");
+
+    await waitUntil(() => popoutWindow.body);
```

Comment:
The PR was about the preview controls (previous, next, popout) not working properly after popout
So I make sure they work before and after popout button is clicked

Reply:
Not working after the popout, so ok for them. Here the comments refer to the `.o_move_next`/`.o_move_previous` that are made before, which are already covered by another test.

This is already covered by another test, so there's no point in testing that again in this test. The test assumes these buttons are working. So only do `.o_move_next` and `.o_move_previous` after doing the popout.

If each test has to test again what other smaller tests did, the newer tests will have hundreds or thousands of assertions. This is a big no-no with code maintenance of test. Keep test focused on the specific aspect being tested. To the extreme each test should have 1 and only 1 precisely relevant assertion.

(of course you can assert more, but the general idea is assert only what's unique and specific to the test at hand, so that we can know that 1K assertions means 1K stuffs are covered in tests, and not just 100 times the 10 same assertions which is very poor coverage value)